internal/commands/accesslist/delete.go

```diff
@@ -0,0 +1,195 @@
+package accesslist
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cli/user"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+const (
+	headerDeleted = "Deleted"
+	headerDetails = "Details"
+)
+
+var (
+	deleteTableHeaders = []string{headerAddress, headerComment, headerDeleted, headerDetails}
+)
+
+type deleteAllowedIPOutputs []deleteAllowedIPOutput
+
+type deleteAllowedIPOutput struct {
+	allowedIP realm.AllowedIP
+	err       error
+}
+type deleteInputs struct {
+	cli.ProjectInputs
+	Addresses []string
+}
+
+// CommandMetaDelete is the command meta for the `accesslist delete` command
+var CommandMetaDelete = cli.CommandMeta{
+	Use:         "delete",
+	Display:     "accesslist delete",
+	Description: "Delete an IP address or CIDR block from the Access List of your Realm app",
+	HelpText: `Removes an existing entry from the Access List of your Realm app. You will be
+prompted to select an IP address or CIDR block if none is provided in the
+initial command.`,
+}
+
+// CommandDelete is the `accesslist delete` command
+type CommandDelete struct {
+	inputs deleteInputs
+}
+
+// Flags is the command flags
+func (cmd *CommandDelete) Flags() []flags.Flag {
+	return []flags.Flag{
+		cli.AppFlagWithContext(&cmd.inputs.App, "to remove an entry in its Access List"),
+		cli.ProjectFlag(&cmd.inputs.Project),
+		cli.ProductFlag(&cmd.inputs.Products),
+		flags.StringSliceFlag{
+			Value: &cmd.inputs.Addresses,
+			Meta: flags.Meta{
+				Name: "ip",
+				Usage: flags.Usage{
+					Description: "Specify the IP address or CIDR block that you would like to delete",
+				},
+			},
+		},
+	}
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	allowedIPs, err := clients.Realm.AllowedIPs(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	selected, err := cmd.inputs.resolveAllowedIP(ui, allowedIPs)
+	if err != nil {
+		return err
+	}
+
+	if len(selected) == 0 {
+		ui.Print(terminal.NewTextLog("No IP addresses or CIDR blocks to delete"))
+		return nil
+	}
+
+	outputs := make(deleteAllowedIPOutputs, len(selected))
+	for i, allowedIP := range selected {
+		err := clients.Realm.AllowedIPDelete(app.GroupID, app.ID, allowedIP.ID)
+		outputs[i] = deleteAllowedIPOutput{allowedIP, err}
+	}
+
+	sort.SliceStable(outputs, func(i, j int) bool {
+		return outputs[i].err != nil && outputs[j].err == nil
```

Comment:
This is how `secrets` are being handled, but I think we can order them based on selection instead. Makes the most sense imo.

Reply:
if its easy enough, making the similar change on `secrets` side would be great too

Reply:
went with the alphanumeric option..

Reply:
out of curiosity what drove that decision?  if its too complex to implement im fine with punting on it for now, but otherwise the more i think about it the more i'm convinced showing the results in the order they were flagged/selected feels the most intuitive here

Reply:
hmm, so actually I think the default is ordering by selection actually, but I haven't looked into how the `&survey.MultiSelect` is inserting them into `selections`.  I can definitely take a deeper look into it. For now I actually removed any sort of sorting/re-ordering the list.

Reply:
cool, lets take that deeper look now and:

* if it defaults to this natural/intuitive sort order, then great, no action needed but let's file a ticket/TODO for `secrets` to operate the same way
* if it doesn't default this way, then still ok, no action needed but let's file a ticket to get both `accesslist` and `secrets` to operate this way

Reply:
https://jira.mongodb.org/browse/REALMC-9795

So the default order for `accesslist` is based on the original list of options (i.e whatever order we gave the user to select from would be maintained after selection as well). I have left the `secrets` selection as it was.